transaction/std/src/transaction_builder.rs

```diff
@@ -492,6 +515,41 @@ impl<FPR: FogPubkeyResolver> TransactionBuilder<FPR> {
         self.fee.token_id
     }
 
+    /// Return low level data to sign and construct transactions with external
+    /// signers
+    pub fn get_signing_data<T: RngCore + CryptoRng>(
+        mut self,
```

Comment:
Why mut? Why does it consume self?

Reply:
Makes sense that it should to me, as it does with build. This should be the end of the flow for the user with this particular instance of the transaction builder.

Reply:
not so with mut, that's a mistake

Reply:
I take that back, it needs to be mutable because of prepare_for_signing

Reply:
I see. As I dig more into this, it appears there is some duplication between `get_signing_data` and `build_with_comparer_internal`. They both start like this:
```
        let (inputs, outputs, output_secrets) = self.prepare_for_signing::<O>().unwrap();

        let tx_prefix = TxPrefix::new(inputs, outputs, self.fee, self.tombstone_block);

        let input_rings = self
            .input_materials
            .into_iter()
            .map(TryInto::try_into)
            .collect::<Result<Vec<InputRing>, _>>()?;

        let message = tx_prefix.hash().0;
 ```

Then, `get_signing_data` proceeds to call `SignatureRctBulletproofs::get_view_only_signing_data`, while `build_with_comparer_internal` calls `SignatureRctBulletproofs::sign`, which in the end also calls `SignatureRctBulletproofs::get_view_only_signing_data` and then does more work.

So I'm wondering if there's a way to simplify things a bit. Something along these lines:
1) `prepare_for_signing` goes away and gets merged into the beginning of `get_signing_data`
2) `build_with_comparer_internal` is modified to call `self.get_signing_data()` and then proceed with signing. This might involve adding some `sign()` method on `ViewOnlySigningData`.
I can make an attempt at doing this if you'd like.

Reply:
Made an attempt: https://github.com/mobilecoinfoundation/mobilecoin/pull/2524